presto-main/src/main/java/io/prestosql/operator/DynamicFilterSourceOperator.java

```diff
@@ -0,0 +1,272 @@
+        {
+            checkState(!closed, "Factory is already closed");
+            return new DynamicFilterSourceOperator(
+                    driverContext.addOperatorContext(operatorId, planNodeId, DynamicFilterSourceOperator.class.getSimpleName()),
+                    dynamicPredicateConsumer,
+                    channels,
+                    planNodeId,
+                    maxFilterPositionsCount,
+                    maxFilterSize);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+            checkState(!closed, "Factory is already closed");
+            closed = true;
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            throw new UnsupportedOperationException("duplicate() is not supported for DynamicFilterSourceOperatorFactory");
+        }
+    }
+
+    private static final int EXPECTED_BLOCK_BUILDER_SIZE = 8;
+
+    private final OperatorContext context;
+    private boolean finished;
+    private Page current;
+    private final Consumer<TupleDomain<String>> dynamicPredicateConsumer;
+    private final int maxFilterPositionsCount;
+    private final long maxFilterSizeInBytes;
+
+    private final List<Channel> channels;
+
+    // May be dropped if the predicate becomes too large.
+    @Nullable
+    private BlockBuilder[] blockBuilders;
+    @Nullable
+    private TypedSet[] valueSets;
+
+    private int filterPositionsCount;
+    private long filterSizeInBytes;
+
+    public DynamicFilterSourceOperator(OperatorContext context, Consumer<TupleDomain<String>> dynamicPredicateConsumer, List<Channel> channels, PlanNodeId planNodeId, int maxFilterPositionsCount, DataSize maxFilterSize)
+    {
+        this.context = requireNonNull(context, "context is null");
+        this.maxFilterPositionsCount = maxFilterPositionsCount;
+        this.maxFilterSizeInBytes = maxFilterSize.toBytes();
+
+        this.dynamicPredicateConsumer = requireNonNull(dynamicPredicateConsumer, "dynamicPredicateConsumer is null");
+        this.channels = requireNonNull(channels, "channels is null");
+
+        this.blockBuilders = new BlockBuilder[channels.size()];
+        this.valueSets = new TypedSet[channels.size()];
+        for (int channelIndex = 0; channelIndex < channels.size(); ++channelIndex) {
+            Type type = channels.get(channelIndex).type;
+            this.blockBuilders[channelIndex] = type.createBlockBuilder(null, EXPECTED_BLOCK_BUILDER_SIZE);
+            this.valueSets[channelIndex] = new TypedSet(type,
+                                                        Optional.empty(),
+                                                        blockBuilders[channelIndex],
+                                                        EXPECTED_BLOCK_BUILDER_SIZE,
+                                                        String.format("DynamicFilterSourceOperator_%s_%d", planNodeId, channelIndex),
+                                                        maxFilterSize);
```

Comment:
I would simply make `TypedSet` unlimited here

Reply:
Sounds good, thanks for the suggestion.